File: mipush_hook/src/main/java/com/xiaomi/push/service/clientReport/PushClientReportManagerAspect.java
package com.xiaomi.push.service.clientReport;

import com.nihility.Hooked;

import org.aspectj.lang.ProceedingJoinPoint;
import org.aspectj.lang.annotation.Around;
import org.aspectj.lang.annotation.Aspect;

@Aspect
public class PushClientReportManagerAspect {
    @Around("execution(* com.xiaomi.push.service.clientReport.PushClientReportManager.collectData(..))")
    public void doNothingToAvoidTracking(final ProceedingJoinPoint joinPoint) {
        Hooked.mark(PushClientReportManager.class.getSimpleName());
    }
}

File: mipush_hook/src/main/java/com/xiaomi/push/service/clientReport/PushClientReportManagerAspect.kt
package com.xiaomi.push.service.clientReport

import com.nihility.Hooked
import org.aspectj.lang.ProceedingJoinPoint
import org.aspectj.lang.annotation.Around
import org.aspectj.lang.annotation.Aspect

@Aspect
class PushClientReportManagerAspect {
    @Around("execution(* com.xiaomi.push.service.clientReport.PushClientReportManager.collectData(..))")
    fun doNothingToAvoidTracking(joinPoint: ProceedingJoinPoint) {
        Hooked.mark(PushClientReportManager::class.java.simpleName)
    }
}
